fdbserver/LogRouter.actor.cpp

```diff
@@ -295,48 +296,68 @@ ACTOR Future<Void> waitForVersion(LogRouterData* self, Version ver) {
 	return Void();
 }
 
+ACTOR Future<Reference<ILogSystem::IPeekCursor>> getPeekCursorData(LogRouterData* self,
+                                                                   Reference<ILogSystem::IPeekCursor> r,
+                                                                   Version startVersion) {
+	state Reference<ILogSystem::IPeekCursor> result = r;
+	state bool useSatellite = SERVER_KNOBS->LOG_ROUTER_PEEK_FROM_SATELLITES_PREFERRED;
+	loop {
+		Future<Void> getMoreF = Never();
+		if (result) {
+			getMoreF = result->getMore(TaskPriority::TLogCommit);
+			++self->getMoreCount;
+			if (!getMoreF.isReady()) {
+				++self->getMoreBlockedCount;
+			}
+		}
+		state double startTime = now();
+		choose {
+			when(wait(getMoreF)) {
+				double peekTime = now() - startTime;
+				self->peekLatencyDist->sampleSeconds(peekTime);
+				self->getMoreTime += peekTime;
+				self->maxGetMoreTime = std::max(self->maxGetMoreTime, peekTime);
+				return result;
+			}
+			when(wait(self->logSystemChanged)) {
+				if (self->logSystem->get()) {
+					result =
+					    self->logSystem->get()->peekLogRouter(self->dbgid, startVersion, self->routerTag, useSatellite);
+					self->primaryPeekLocation = result->getPrimaryPeekLocation();
+					TraceEvent("LogRouterPeekLocation", self->dbgid)
+					    .detail("LogID", result->getPrimaryPeekLocation())
+					    .trackLatest(self->eventCacheHolder->trackingKey);
+				} else {
+					result = Reference<ILogSystem::IPeekCursor>();
+				}
+				self->logSystemChanged = self->logSystem->onChange();
+			}
+			when(wait(result ? delay(SERVER_KNOBS->LOG_ROUTER_PEEK_SWITCH_DC_TIME) : Never())) {
+				// Peek has become stuck for a while, trying switching between primary DC and satellite
+				TraceEvent("LogRouterSlowPeek", self->dbgid).detail("NextTrySatellite", !useSatellite);
+				useSatellite = !useSatellite;
+				result =
+				    self->logSystem->get()->peekLogRouter(self->dbgid, startVersion, self->routerTag, useSatellite);
+				self->primaryPeekLocation = result->getPrimaryPeekLocation();
+				TraceEvent("LogRouterPeekLocation", self->dbgid)
+				    .detail("LogID", result->getPrimaryPeekLocation())
```

Comment:
s/LogID/PeekLocation ?

Reply:
This is the TLog ID, so I felt `LogID` is accurate and easier to understand.